### PR TITLE
Documenting some recomendations to add testing with Pytest

### PR DESCRIPTION
**Pull request by czoido**
Changelog: Omit
Docs: Omit

Add documentation on some general guidelines when adding tests to Conan: naming of files and tests, marks that should be added, tests folder organization...

### CODE REVIEW

File: conans/test/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+discovery:
+- Name your Python test files ending in `_test.py`.
+
+```
+test
+├── README.md
+├── conftest.py
+├── unittests
+│   ├── __init__.py
+│   ├── basic_build_test.py
```
Comment (memsharded) [inline conans/test/README.md]: Better start names with ``test_``. It makes evident when something is not named (and not running)

**Review by memsharded — APPROVED** [review]
LGTM with that minor change about filenames

File: conans/test/README.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+```
+
+If the test needs any of those tools to run it should be marked as using that tool and moved to the `conans/test/functional` folder.
+
+### Parametrizing tests
+
+Please, if you need to run several combinations of the same testing code use parameterization. You can use the builtin `pytest.mark.parametrize` decorator to enable parametrization of arguments for a test function:
+
+```python
+@pytest.mark.parametrize("use_components", [False, True])
```
Comment (jgsogo) [inline conans/test/README.md]:
```suggestion
    @pytest.mark.parametrize("use_components", [False, True])
```

File: conans/test/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+Pytest follows this [convention](https://docs.pytest.org/en/stable/goodpractices.html) for test
+discovery:
+- Name your Python test files starting in `test_`.
+
+```
+test
+├── README.md
+├── conftest.py
+├── unittests
```
Comment (jgsogo) [inline conans/test/README.md:37]:
```suggestion
├── integration
```

Related to `unittests`, which is the example here, but it was something I wanted to comment in this PR.

This is me **completely opinionated about this**: I REALLY like the tests under `unittests` to follow EXACTLY the same directory/files structure as the codebase, why?
 * no doubts where to add a test
 * no doubts where to check if a function is tested before writing one more test (was it already tested?).
 * when you are refactoring something, you only need to run one folder while changing the implementation.

We can talk about this later, but then I'd change the example here and show a file from `functional` or `integration`

Comment (memsharded) [reply conans/test/README.md:37]: I agree, unittests should follow the same layout as the prod codebase.

**Review by memsharded — APPROVED** [review]

**Pull request merged**